flight_phase_files/HALO/EUREC4A_HALO_Flight-Segments_20200215.yaml

```diff
@@ -297,7 +296,7 @@ segments:
   - circle 6 Only 7 dropsondes
   - SAM HALO-0215_s32 sonde added manually, released before traffic diversion
   segment_id: HALO-0215_o4
-  start: 2020-02-15 21:50:20
+  start: 2020-02-15 21:48:22
```

Comment:
This changes the timeframe of the segment. I've had some fear that this will eventually happen... But I don't have a sound idea about all implications and how to handle this properly. The issue is that from what I saw before, we usually don't refer to a segment at a given revision of the flight segmentation files, but rather just to a segment using a given segment id. We don't yet have fully specified what the "core" of a segment is, but one idea of defining segments has been that they allow to uniquely refer to certain parts of a flight. Changing the start and end times of the segment feels like very close to ripping apart the idea of a unique reference to a part of the flight.

So:
* could you elaborate on why this is necessary?
* should we probably change the segment id to avoid confusion?

Reply:
I think there was an error in the original YAML file. The circling period was starting at 21:50:20, but the first circle in the circling period has start timestamp of 21:48:22. No irregularity also is reported. Therefore, the circling start time is corrected now to 21:48:22. The segment is essentially the same, one of its properties has changed and therefore, I don't think the segment id needs to be changed. I agree that changing the start and end times does not correspond to the idea of the unique reference, but this can be considered as correcting some erroneous data, right? I hope I am clear. Tried to check in a better way with the `reports.py` but unable to run that because of that same error with something about the `ScannerError: while scanning a simple key`...

Reply:
Just checked with the HTML generated by `reports.py`. There was an error in the original file. Circle c5 started at 21:48:22, and circling o4 started at 21:50:20. The latter was likely changed because there is a slight altitude change (pitch can also be seen as slightly higher). But if this is to be considered correct, then the same should also have been followed for c5. The difference between the two causes the difference of a dropsonde. I recommend we make this change and start o4 at 21:48:22.  A significant duration before the end time also has major changes in altitude and pitch. Therefore the original "argument" for the latter timestamp is not very robust anyway.

![Flight track details for HALO-0215_o4](https://user-images.githubusercontent.com/46085279/114766570-76c46080-9d5e-11eb-9978-643f94a21daf.png)


![Zoomed-in view of pitch and altitude at start and end](https://user-images.githubusercontent.com/46085279/114766727-a8d5c280-9d5e-11eb-979c-f17300d2ecd4.png)

Reply:
Regarding the reordering, I will change the ordering back to the original state. What happened to the changes in the sonde flags (also for sondes that came into or went out of existence) is that Level-1 files changed. Therefore, some data which is input for the QC before Level-2 also changed. This caused the changes. But I have maintained the same, original sonde_ids. One sonde that got added was given the last `s` number of the flight instead of whatever sequence it was coming in (I think 27 or something), and one sonde that got removed also took with it the  `sonde_id` that it was originally given. This means that that sonde id is not present in the FPS files (or for that matter in JOANNE) anymore.

Reply:
So the thing is, there [this commit](https://github.com/eurec4a/flight-phase-separation/commit/10eac6c4e7815763482f9d13bf5f3237a6f0e84d#diff-fbc7d92109ff22de41edcbd366f47c394afe83cc38a455adf86a33d81bdd7d53L299-R297) which intentionally does the reverse of this start time modification. I can't recall why we chose to do so and the commit message isn't very expressive either... Maybe we should ask the original author?

Reply:
Yes, that's what I mean. Maybe it was changed in [that commit](https://github.com/eurec4a/flight-phase-separation/commit/10eac6c4e7815763482f9d13bf5f3237a6f0e84d#diff-fbc7d92109ff22de41edcbd366f47c394afe83cc38a455adf86a33d81bdd7d53L299-R297) to avoid the timestamps when pitch was high and altitude was changing. But then this means that the same should be followed for segment c5. So, either one of the start timestamps should change, because one would expect that `circle` segments belong to the `circling` segment. I would argue that consistency among segments should supersede persistence among versions. Marc is the original author (`dp`) for the flight. We could ask him...

Reply:
So, one workaround to go ahead without changing the timestamps can be that `HALO-0215_s33` is attributed to the circling segment `o4` manually, and we attach a SAM irregularity for the segment. This is a fix in order to not meddle with the timestamps present in the Awesome Albatross FPS files. We would still need to discuss if it is okay to have a circle partially within the circling time, but maybe this can be done in a different PR. 

My PoV is that for syncing FPS with JOANNE, the timestamps of the segments don't matter. Only the sondes attributed to the segments do. Therefore, if this PR can go ahead, I can update Jo-Level-4. And later on, if we do decide to change either the timestamp for `o4` or `c5` (which I think we should), then it would still not affect how JOANNE behaves. 

If that seems okay to you, I can do this SAM workaround that I suggested above and change the timestamp to what was in the original.

Reply:
Yes, that's probably a good idea. In general, I would agree with you that the timestamps should probably be adjusted. I just don't want to rush that before we really have a good reason to do so and we understand the implications.

Reply:
Yes, agreed.

Reply:
So, I pushed another commit reverting back to the original timestamp (i.e. 21:50:xx) and adding `s33` as SAM as per the workaround discussed above, but I think even with the original timestamp, running `attach_sondes.py` would not have included s33 in the segment and a SAM for `s33` should have been included in the AA FPS files. It probably didn't show up as a failure in the test, because the segment already contains a SAM tag for `s32`. 

What I speculate is that after the timestamp was changed with [this commit](https://github.com/eurec4a/flight-phase-separation/commit/10eac6c4e7815763482f9d13bf5f3237a6f0e84d#diff-fbc7d92109ff22de41edcbd366f47c394afe83cc38a455adf86a33d81bdd7d53L299-R297), the `attach_sondes.py` was not run on the file. 

So, in all, two suggestions:
(a) we should change either one of the start timestamps in `o4` or `c5`, otherwise include an irregularity in both and mention the reason for keeping this different
(b) the SAM irregularities should recorded once per instance of any sonde being attributed manually, and the tests should check for this.